.github/workflows/build.yml

```diff
@@ -147,36 +123,24 @@ jobs:
           git log --oneline -n$(( COMMITS + CONTEXT ))
       - name: Install nix
         uses: cachix/install-nix-action@v30
-        with:
-          extra_nix_config: |
-            connect-timeout = 5
-            system-features = nixos-test benchmark big-parallel kvm
-            builders-use-substitutes = true
-            builders = @/etc/nix/machines
-            log-lines = 100
-            # do not build anything locally, delegate all builds to remote builders:
-            max-jobs = 0
-            # allow nix build to use the binary cache(s) configured in ghaf flake:
-            trusted-users = runner
-            accept-flake-config = true
-      - name: Configure remote builder
+      - name: Prepare build
         run: |
-          sudo sh -c "umask 377; echo '${{ secrets.BUILDER_SSH_KEY }}' >/etc/nix/id_builder_key"
+          sh -c "umask 377; echo '${{ secrets.BUILDER_SSH_KEY }}' >builder_key"
           sudo sh -c "echo '${{ vars.BUILDER_SSH_KNOWN_HOST }}' >>/etc/ssh/ssh_known_hosts"
-          sudo sh -c "echo '${{ vars.BUILDER_MACHINE_CONFIG }}' >/etc/nix/machines"
-      - name: Install cachix
-        run: |
-          nix-env -iA cachix -f https://cachix.org/api/v1/install
-          echo "Using cachix version:"
-          cachix --version
       - name: Build ${{ matrix.arch }}.${{ matrix.target }}
         run: |
-          if [ "${{ secrets.CACHIX_AUTH_TOKEN }}" == "" ]; then
-            echo "::error::Missing CACHIX_AUTH_TOKEN, will not build"
-            exit 1
+          if [ "${{ matrix.arch }}" == "x86_64-linux" ]; then
+            BUILDER='${{ vars.BUILDER_X86 }}'
+          elif [ "${{ matrix.arch }}" == "aarch64-linux" ]; then
+            BUILDER='${{ vars.BUILDER_AARCH }}'
           else
-            echo "Running nix build, with cachix watch-exec"
-            cachix authtoken ${{ secrets.CACHIX_AUTH_TOKEN }}
-            cachix watch-exec -v ghaf-dev -- \
-              nix build .#packages.${{ matrix.arch }}.${{ matrix.target }}
+            echo "::error::Unknown architecture: '${{ matrix.arch }}'"
+            exit 1
           fi
+          NIX_FAST_BUILD_GITREF="1775c732"
```

Comment:
Do we need to pin the version?

Reply:
I would prefer that we pin the version. That way, we would be in control when/if we decide to update the `nix-fast-build` version we use, and would not be caught off guard on updates.

Reply:
Sure. I'll try to remember to bump it on every flake lick update, to keep in sync with the dev shell